### PR TITLE
Add documentation for new facet behavior

### PR DESCRIPTION
**Pull request by OliverSkroblin**

### CODE REVIEW

File: source/developers-guide/elasticsearch/index.md [hunk 1]
```diff
@@ -1908,3 +1908,79 @@ class SalesSortingHandler implements HandlerInterface
 
 Like the other handlers, the `SalesSortingHandler` handles a `CriteriaPart`, so it implements the `Shopware\Bundle\SearchBundleES\HandlerInterface`.
 To sort the search result, the `ONGR\ElasticsearchDSL\Sort\Sort` class can be added to the provided `ONGR\ElasticsearchDSL\Search`.
+
+## Shopware 5.3 changes
+With Shopware 5.3 a new filter behavior was implemented which has to display only filters which can be combined with the current user conditions.
```
Comment (ndzoesch) [inline source/developers-guide/elasticsearch/index.md]: Could you rephrase? Something like `A new filter behaviour was implemented with shopware 5.3, now filters are only displayed if they can be combined with the user conditions that are currently in effect.` sounds better imho.

File: source/developers-guide/elasticsearch/index.md [hunk 2]
```diff
@@ -1908,3 +1908,79 @@ class SalesSortingHandler implements HandlerInterface
...
+In the elastic search implementation the filter behavior is controlled by the condition handlers. By adding an query as `post filter`, facets are not affected by other filters.
```
Comment (ndzoesch) [inline source/developers-guide/elasticsearch/index.md]: ~~an query~~ a query

File: source/developers-guide/elasticsearch/index.md [hunk 3]
```diff
@@ -1908,3 +1908,79 @@ class SalesSortingHandler implements HandlerInterface
...
+This behavior is checked over the `Criteria->hasBaseCondition` statement:
```
Comment (ndzoesch) [inline source/developers-guide/elasticsearch/index.md]: ~~over~~ via

File: source/developers-guide/elasticsearch/index.md [hunk 4]
```diff
@@ -1908,3 +1908,79 @@ class SalesSortingHandler implements HandlerInterface
...
+    if ($criteria->hasBaseCondition($criteriaPart->getName())) {
+        $search->addFilter(new TermQuery('active', 1));
+    } else {
+        $search->addPostFilter(new TermQuery('active', 1));
+    }
+}
+
+```
+This behavior is now controlled in the `\Shopware\Bundle\SearchBundleES\ProductNumberSearch`. To support the new filter mode, each condition handler has to implement the `\Shopware\Bundle\SearchBundleES\PartialConditionHandlerInterface`.
+It is possible to implement this interface beside the original `\Shopware\Bundle\SearchBundleES\HandlerInterface`.
```
Comment (ndzoesch) [inline source/developers-guide/elasticsearch/index.md]: ~~beside~~ alongside

File: source/developers-guide/shopware-5-search-bundle/index.md [hunk 5]
```diff
@@ -231,3 +231,85 @@ In most cases, only a list of product numbers is not enough for the store front.
 The `Shopware\Bundle\SearchBundle\ProductSearch` class is a decorator, which allows the execution of a product number search, whose result sets contains a list of `Shopware\Bundle\StoreFrontBundle\Struct\ListProduct` structs.
 A `ListProduct` struct contains all required data for a product to be display in the store front listings.
 Internally, this decorator first executes the `ProductNumberSearch` and then converts the result using the configured `Shopware\Bundle\StoreFrontBundle\Service\ListProductServiceInterface` from the DI container.
+
+## Shopware 5.3 changes
+With Shopware 5.3, a new interface implemented to handle facets: `\Shopware\Bundle\SearchBundleDBAL\PartialFacetHandlerInterface`. Each facet handler had to revert the provided criteria by their own to remove customer conditions. This behaviour is now handled in the `\Shopware\Bundle\SearchBundleDBAL\ProductNumberSearch::createFacets`
```
Comment (ndzoesch) [inline source/developers-guide/shopware-5-search-bundle/index.md]: "With Shopware 5.3, a new interface **is**  implemented to handle facets", or better imho: "Shopware 5.3 implements a new interface to handle facets"

File: source/developers-guide/shopware-5-search-bundle/index.md [hunk 6]
```diff
@@ -231,3 +231,85 @@ In most cases, only a list of product numbers is not enough for the store front.
...
+if (!interface_exists('\Shopware\Bundle\SearchBundleDBAL\PartialFacetHandlerInterface')) {
+    interface PartialFacetHandlerInterface
+    {
+    }
+}
+
+class MyHandlerClass implements FacetHandlerInterface, PartialFacetHandlerInterface
+{
+    /**
+     * {@inheritdoc}
```
Comment (ndzoesch) [inline source/developers-guide/shopware-5-search-bundle/index.md]: I would omit the "inheritdoc"-blocks, since it adds no value to the documentation.

File: source/developers-guide/elasticsearch/index.md (hunk 2 above)
Comment (OliverSkroblin) [inline source/developers-guide/elasticsearch/index.md]: changed

File: source/developers-guide/elasticsearch/index.md (hunk 3 above)
Comment (OliverSkroblin) [inline source/developers-guide/elasticsearch/index.md]: changed

File: source/developers-guide/elasticsearch/index.md (hunk 4 above)
Comment (OliverSkroblin) [inline source/developers-guide/elasticsearch/index.md]: changed
